src/inausoft.netCLI/netCLI.cs

```diff
@@ -0,0 +1,99 @@
+﻿using inausoft.netCLI.Commands;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace inausoft.netCLI
+{
+    public static class netCLI
+    {
+        /// <summary>
+        /// Setups CLI flow by adding and configuring <see cref="RootCommandHandler"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="SetupHelpCommand"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddCLI(this IServiceCollection services, Action<CLIConfiguration> setup)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException($"{nameof(services)} cannot be null.");
+            }
+
+            CLIConfiguration configuration = new CLIConfiguration();
+
+            setup(configuration);
+
+            services.AddSingleton(configuration);
+
+            configuration._commandMap.Values.ToList().ForEach(it =>
+            {
+                services.AddSingleton(it);
+            });
+
+            return services;
+        }
+
+        /// <summary>
+        /// Runs <see cref="ICommandHandler"/> for command specified in args./>
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="args">Application's command line arguments.</param>
+        /// <returns></returns>
+        public static int RunCLI(this IServiceProvider serviceProvider, string[] args)
```

Comment:
Do we have any behaviour when user does not write any arguments? For example , route to help command?

Reply:
This would need to be added. A either imagine that you could go into infinite loop where you can execute existing commends without calling .exe again - or just add some bool parameter into maping function:
```cs
public CLIConfiguration Map<T1, T2>(bool setAsDefault = false) ...
```

But this is a subject of another PR.

Reply:
Oh I commented on the same thing below I guess... Nice

Reply:
I saw that @matiduck opened #9 let's continue this disscusion there.